### ISSUE TITLE
The  calculation of max_train_steps is not correct

### ISSUE THREAD

**Issue by Rokurolize** [derrian-distro/LoRA_Easy_Training_Scripts#8]
Example:
22 images * 10 repetitions = 220
Batch size 50
If I want to train with 4 epochs, I have the following:
(220 / 50) * 4 = 17.6
https://github.com/derrian-distro/LoRA_Easy_Training_Scripts/blob/e981b5433c2d2dd6ff4dbd317a6e99329d0b6e56/lora_train_command_line.py#L215-L241
In this case, --max_train_steps=17, but when I run the training script, it says the number of batches per epoch is 7, and 4 epochs are 28 steps, so it does not reach the number of epochs I expected.
From what I have heard, the actual process calculates the number of data for each bucket, which causes discrepancies.
![](https://user-images.githubusercontent.com/1701388/216727701-4eec613e-282e-443e-abaf-855fb26e6ede.png)

Looking at kohya's script, I found this code, so I think we can pass the epoch number as it is.
https://github.com/kohya-ss/sd-scripts/blob/9682772b09a37d42dbf9963153a7ceaabcff6182/train_network.py#L219-L223
If --max_train_epochs=4 is added:
![](https://user-images.githubusercontent.com/1701388/216728759-b329e045-d61c-4143-9573-62c09fae9738.png)

**Comment by derrian-distro** [comment]
yeah, I was looking through the code and found that just a bit ago, I was planning on making that change in the next update, saves me the trouble also. Thanks for bringing it to me though.

**Comment by derrian-distro** [comment]
Hi Rokurolize, I changed up the way max steps are calc'd, so now they just use built in max_train_epoch and max_train_step setups. I kept in the calc code for when people use the warmup ratio, but that's about all it's used for now. I'm gonna close this now, reopen it if you have any other issues with the calc. Thanks
